src/main/java/lotto/controller/LottoController.java

```diff
@@ -0,0 +1,84 @@
+package lotto.controller;
+
+import java.util.List;
+import lotto.constant.LottoRanking;
+import lotto.generator.RandomUniqueListGenerator;
+import lotto.model.Lotto;
+import lotto.model.LottoGameManager;
+import lotto.model.LottoOwner;
+import lotto.model.LottoResult;
+import lotto.model.PurchasePrice;
+import lotto.model.WinningLotto;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+public class LottoController {
+    private final InputView inputView = new InputView();
+    private final OutputView outputView = new OutputView();
+
+    public void run() {
+        LottoOwner lottoOwner = getLottoOwner();
+        outputView.printLottosInfo(lottoOwner.getLottosInfo());
+        Lotto winningNumbers = getWinningNumbersUntilValid();
+        outputView.printSpace();
+        WinningLotto winningLotto = getWinningLottoUntilValidBonusNumber(winningNumbers);
+        LottoGameManager lottoGameManager = LottoGameManager.createGameManager(lottoOwner, winningLotto);
+        lottoGameManager.matchLottosWithWinningLotto();
+        printWinningStatistics(lottoGameManager);
+        outputView.printRateOfReturn(lottoGameManager.getRateOfReturn());
+    }
+
+    private LottoOwner getLottoOwner() {
+        PurchasePrice purchasePrice = getPurchasePriceUntilValid();
+        List<Lotto> randomLottos = Lotto.getAutoGeneratedLottos(purchasePrice, new RandomUniqueListGenerator());
+        LottoResult lottoResult = LottoResult.createLottoResult(LottoRanking.convertValuesIntoCounters());
+        return LottoOwner.createLottoOwner(purchasePrice, randomLottos, lottoResult);
+    }
+
+    private PurchasePrice getPurchasePriceUntilValid() {
+        PurchasePrice purchasePrice;
+        while (true) {
+            try {
+                outputView.printEnterPurchaseAmount();
+                purchasePrice = PurchasePrice.createPurchase(inputView.getPurchaseAmount());
+                break;
+            } catch (IllegalArgumentException e) {
+                outputView.printText(e.getMessage());
+            }
+        }
+        return purchasePrice;
+    }
```

Comment:
```suggestion
    private PurchasePrice getPurchasePriceUntilValid() {
        while (true) {
            try {
                outputView.printEnterPurchaseAmount();
                PurchasePrice purchasePrice = PurchasePrice.createPurchase(inputView.getPurchaseAmount());
                return purchasePrice;
            } catch (IllegalArgumentException e) {
                outputView.printText(e.getMessage());
            }
        }
    }
```

break문도 좋지만 검증이 되면 바로 리턴해줘도 좋을 것 같아요 😄

Reply:
저도 같은 생각이었는데 이미 적어두셨네요! 재귀 호출을 피하신 부분은 좋다고 생각합니다 👍🏼

Reply:
try 내부에서 return 반환이 가능했군요! 제대로 배워갑니다 🤗